main/exercise/ptest_agree_scale.class.php

```diff
@@ -0,0 +1,346 @@
+ *
+ * @author Jose Angel Ruiz
+ *
+ * @package chamilo.exercise
+ */
+class PtestAgreeScale extends Question
+{
+    public $typePicture = 'ptscale.png';
+    public $explanationLangVar = 'PtestAgreeScale';
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->type = QUESTION_PT_TYPE_AGREE_SCALE;
+        $this->isContent = $this->getIsContent();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createAnswersForm($form)
+    {
+        // Getting the exercise list
+        /** @var Exercise $obj_ex */
+        $obj_ex = Session::read('objExercise');
+
+        $editor_config = [
+            'ToolbarSet' => 'TestProposedAnswer',
+            'Width' => '100%',
+            'Height' => '125',
+        ];
+
+        // Categories options select
+        $category = new PTestCategory();
+        $categoriesList = $category->getCategoryListInfo($obj_ex->selectId());
+        $categoriesOptions = [null => get_lang('None')];
+        foreach ($categoriesList as $categoryItem) {
+            $categoriesOptions[$categoryItem->id] = (string) $categoryItem->name;
+        }
+
+        //this line defines how many questions by default appear when creating a choice question
+        // The previous default value was 2. See task #1759.
+        $nb_answers = isset($_POST['nb_answers']) ? (int) $_POST['nb_answers'] : count($categoriesList);
+        $nb_answers += (isset($_POST['lessAnswers']) ? -1 : (isset($_POST['moreAnswers']) ? 1 : 0));
+
+        $html = '<table class="table table-striped table-hover">
+            <thead>
+                <tr style="text-align: center;">
+                    <th width="5%">'.get_lang('Number').'</th>
+                    <th width="50%">'.get_lang('Answer').'</th>
+                    <th width="45%">'.get_lang('PtestCategory').'</th>
+                </tr>
+            </thead>
+            <tbody>';
+
+        $form->addHeader(get_lang('Answers'));
+        $form->addHtml($html);
+
+        $defaults = [];
+        if (!empty($this->id)) {
+            $answer = new Answer($this->id);
+            $answer->read();
+            if ($answer->nbrAnswers > 0 && !$form->isSubmitted()) {
+                $nb_answers = $answer->nbrAnswers;
+            }
+        }
+        $form->addElement('hidden', 'nb_answers');
+
+        //$temp_scenario = [];
+        if ($nb_answers < 1) {
+            $nb_answers = 1;
+            echo Display::return_message(
+                get_lang('YouHaveToCreateAtLeastOneAnswer')
+            );
+        }
+
+        for ($i = 1; $i <= $nb_answers; $i++) {
+            $form->addHtml('<tr>');
+            if (isset($answer) && is_object($answer)) {
+                $defaults['answer['.$i.']'] = isset($answer->answer[$i]) ? $answer->answer[$i] : '';
+                $defaults['ptest_category['.$i.']'] = isset($answer->ptest_category[$i]) ? $answer->ptest_category[$i] : 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 123 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)